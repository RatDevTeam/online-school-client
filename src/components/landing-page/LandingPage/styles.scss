@import '~Styles/variables';
.landing {
  &__container {
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: repeat(4, auto);
    padding: 0 100px;
    color: $dark-blue;
    > div {
      margin-top: 100px;
    }
    grid-template-areas:
    "landing-title"
    "advantages"
    "how-to-study"
    "free-webinar"
    "feedback-carousel"

  }
  &__title {
    grid-area: landing-title;
    display: flex;
    &__left {
      display: flex;
      flex-direction: column;
      flex-basis: 50%;
      justify-content: center;
      padding-left: 10%;
      box-sizing: border-box;

      h2{
        font-size: 36px;
        line-height: 49px;
        margin: 0;
        max-width: 340px;
      }
      p{
        color: $grey-text;
      }
      button{
        background-color: $accent-red;
        color: $white;
        padding: 0 3em;
        text-align: center;
        font-size: 18px;
        font-weight: bold;
        height: 50px;
        width: 20em;
      }
    }
    &__img {
      background: no-repeat center/100% $img-likbez-main;
      display: flex;
      flex: 1;
      min-height: 450px;
      height: 100%;
    }
  }

  &__advantages {
    grid-area: advantages;
    display: flex;
    flex-direction: column;
    &__title {
      text-align: center;
      font-size: 48px;
      span {
        color: $saturated-red;
      }
    }
    &__body {
      display: flex;
      width: 100%;
      &__box {
        display: flex;
        flex-direction: column;
        flex: 1;
        &.right {
          justify-content: center;
          margin-left: 40px;
        }
        &__el {
          display: flex;
          flex-direction: row;
          width: 100%;
          margin-bottom: 40px;
          &__img {
            background: no-repeat center/100% $img-tick;
            min-width: 50px;
            height: 50px;
            margin: 10px 10px 0 0;
          }
          &__text {
            display: flex;
            flex-direction: column;
            span {
              color: $saturated-red;
            }
            ul {
              color: $grey-text;
              margin: 0 0 0 10px;
              padding-left: 10px;
            }

            li {
              margin: 10px 0;
            }
          }
        }
      }
    }
  }


  &__free-webinar{
    grid-area: free-webinar;
    transform: translateZ(-1px);
    background-image: $img-free-webinar;
    margin-left: -100px;
    margin-right: -100px;
    &__title{
      h1{
        color: $white;
        font-size: 48px;
        margin-top: 100px;
        text-align: center;
      }
    }
    &__body {
      display: flex;
      flex-direction: row;
      margin-bottom: 100px;
      &__left {
        background: rgba(255, 255, 255, 0.5);
        padding: 3em 100px;

        button {
          background-color: $accent-red;
          color: $white;
          padding: 0 3em;
          text-align: center;
          font-size: 18px;
          font-weight: bold;
          height: 50px;
          width: 15em;
        }

        p {
          color: $dark-blue;
          font-weight: bold;
          font-size: 24px;
        }
      }

      &__right {
        display: flex;
        flex-direction: column;
        justify-content: center;
        margin-left: 40px;
        h1 {
          color: $white;
          font-size: 48px;
        }
        ul {
          margin-left: 10px;
          padding-left: 10px;
        }
        li {
          list-style-type: none;
          color: $white;
          font-weight: bold;
          font-size: 24px;
        }

        li:before {
          content: 'â–º';
          margin-left: -20px;
          margin-right: 10px;
          color: $accent-red;
        }

      }
    }
  }


  &__how-to-study {
    grid-area: how-to-study;
    background-color: $white;
    display: flex;
    flex-direction: column;

    &__title {
      h1 {
        font-size: 48px;
        font-weight: bold;
        text-align: center;
        margin: 0;
      }
      span {
        color: $accent-red;
      }
    }

    &-body {
      display: flex;
      flex-direction: row;
      align-items: center;
      &__text {
        display: flex;
        flex-direction: column;
        flex: 1;
        max-width: 500px;
        color: $dark-grey-text;
        font-weight: normal;
        font-size: 18px;
        padding: 0 2em;
      }
      &__img {
        background: no-repeat center/100% $img-likbez-courses;
        flex: 1;
        height: 100%;
        min-height: 450px;
      }
    }
  }
}

@media (max-width: 1024px) {
  .landing {
    &__container {
      padding: 0 20px;
    }

    &__title {
      flex-direction: column;
      align-items: center;
      margin-top: 70px;

      &__left {
        align-items: center;
        padding: 0;

        h2 {
          max-width: 100%;
        }
      }

      &__img {
        width: 100%;
        max-width: 600px;
      }
    }

    &__how-to-study-body {
      flex-direction: column;
      &__img {
        max-width: 600px;
        width: 100%;
      }
      &__text {
        width: 90%;
        padding: 0;
      }
    }
    &__free-webinar {
      margin-left: -20px;
      margin-right: -20px;
      &__body {
        flex-direction: column;
        &__right {
          align-self: center;
        }
      }
    }
  }
}

@media (max-width: 768px) {
  .landing {
    &__title__left {
    }
    h2 {
      max-width: 340px;
    }
    &__advantages__body {
      flex-direction: column;
      &__box {
        &.right {
          margin: 0;
          justify-content: flex-start;
        }
      }
    }
  }
}

@media (max-width: 480px) {
  .landing {
    &__title {
      &__left {
        button {
          font-size: 14px;
        }
      }
      &__img {
        min-height: 300px;
      }
    }
    &__free-webinar__body__left {
      padding: 3em 10%;
      button {
        font-size: 14px;
      }
    }
    &__how-to-study-body__img{
      min-height: 300px;
    }
  }
}

@media (max-width: 360px) {
  .landing {
    &__container {
      h1 {
        font-size: 32px;
      }
    }
  }
}
