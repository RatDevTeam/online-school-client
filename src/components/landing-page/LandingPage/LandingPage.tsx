import React, { useEffect } from 'react';
import AOS from 'aos';
import Feedback from '../FeedbackCarousel';
import advantages from '../../../utils/advantages';

import './styles.scss';

interface ILandingPageProps {}

const LandingPage: React.FC<ILandingPageProps> = () => {
	useEffect(() => {
		AOS.init({
			delay: 300,
		});
	});
	return (
		<>
			<div className="landing__container">
				<div className="landing__title">
					<div className="landing__title__left" data-aos="fade-right">
						<h2> Подготовка к ЕГЭ в онлайн формате</h2>
						<p>
							Преподаватели и тьюторы ЛикБез нацелены на то,
							<br /> чтобы ты максимально эффективно
							<br /> подготовился к экзамену
						</p>
						<button type="button">Я сдам на 100!</button>
					</div>
					<div className="landing__title__img" data-aos="fade-left" />
				</div>
				<div className="landing__advantages">
					<h1 className="landing__advantages__title">
						Подготовим к <span>ЕГЭ</span>
					</h1>
					<div className="landing__advantages__body">
						<div className="landing__advantages__body__box">
							{advantages.left.map(
								(a: { id: number, title: string, list: string[] }) => (
									<div className="landing__advantages__body__box__el">
										<div className="landing__advantages__body__box__el__img"></div>
										<div className="landing__advantages__body__box__el__text">
											<h2>
												<span>Без</span> {a.title}
											</h2>
											<ul>
												{a.list.map((l: string) => (
													<li>{l}</li>
												))}
											</ul>
										</div>
									</div>
								)
							)}
						</div>
						<div className="landing__advantages__body__box right">
							{advantages.right.map(
								(a: { id: number, title: string, list: string[] }) => (
									<div className="landing__advantages__body__box__el">
										<div className="landing__advantages__body__box__el__img" />
										<div className="landing__advantages__body__box__el__text">
											<h2>
												<span>Без</span> {a.title}
											</h2>
											<ul>
												{a.list.map((l: string) => (
													<li>{l}</li>
												))}
											</ul>
										</div>
									</div>
								)
							)}
						</div>
					</div>
				</div>
				<div
					className="landing__how-to-study"
					data-aos="fade-up"
					data-aos-duration="500"
				>
					<div className="landing__how-to-study__title">
						<h2>
							Как проходят занятия<span>?</span>
						</h2>
					</div>
					<div className="landing__how-to-study-body">
						<div className="landing__how-to-study-body__img" />
						<div className="landing__how-to-study-body__text">
							<p>
								Занятия проходят онлайн. Для каждого занятия есть скрипт, чтобы
								было проще вести конспект, а информация лучше усваивалась.
							</p>
							<p>
								Длительность: от часа до полутора часов (в зависимости от темы).
								Начало всегда по московскому времени. Но не переживай, если ты
								не успел посетить занятие: мы всегда сохраняем видеозапись.
							</p>
							<p>
								Личный тьютор будет помогать тебе, а преподаватель ответит на
								вопросы
							</p>
						</div>
					</div>
				</div>
				<div className="landing__free-webinar">
					<div className="landing__free-webinar__title">
						<h2>Посети первое занятие бесплатно</h2>
					</div>
					<div className="landing__free-webinar__body">
						<div className="landing__free-webinar__body__left">
							<p>
								Подпишись на нас вКонтакте
								<br />и получи приглашение на первое занятие. <br /> Пора
								готовиться!
							</p>
							<button type="button">Подписаться</button>
						</div>
						<div className="landing__free-webinar__body__right">
							<h3>Тебе доступны курсы</h3>
							<ul>
								<li>Английский язык. ЕГЭ</li>
								<li>История. ЕГЭ</li>
								<li>Обществознание. ЕГЭ</li>
							</ul>
						</div>
					</div>
				</div>
			</div>
			<h2>Отзывы учеников</h2>
			<Feedback />
		</>
	);
};
export default LandingPage;
