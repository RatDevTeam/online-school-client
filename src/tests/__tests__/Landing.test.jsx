import React from 'react';
import renderer from 'react-test-renderer';
import LandingPage from '../../components/landing-page/LandingPage/LandingPage';

it('корректно рендерит лендинг ликбез', () => {
	const tree = renderer.create(<LandingPage/>).toJSON();
	expect(tree).toMatchSnapshot();
});
