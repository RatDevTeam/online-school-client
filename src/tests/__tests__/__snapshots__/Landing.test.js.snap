// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`корректно рендерит лендинг ликбез 1`] = `
<div
  className="landing__container"
>
  <div
    className="landing__title"
  >
    <div
      className="landing__title__left"
      data-aos="fade-right"
    >
      <h2>
         Подготовка к ЕГЭ в онлайн формате
      </h2>
      <p>
        Преподаватели и тьюторы ЛикБез нацелены на то,
        <br />
         чтобы ты максимально эффективно
        <br />
         подготовился к экзамену
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Я сдам на 100!
      </button>
    </div>
    <div
      className="landing__title__img"
      data-aos="fade-left"
    />
  </div>
  <div
    className="landing__advantages"
  >
    <h1
      className="landing__advantages__title"
    >
      Подготовим к 
      <span>
        ЕГЭ
      </span>
    </h1>
    <div
      className="landing__advantages__body"
    >
      <div
        className="landing__advantages__body__box"
      >
        <div
          className="landing__advantages__body__box__el"
        >
          <div
            className="landing__advantages__body__box__el__img"
          />
          <div
            className="landing__advantages__body__box__el__text"
          >
            <h2>
              <span>
                Без
              </span>
               
              воды и лишней информации
            </h2>
            <ul>
              <li>
                Материал на вебинарах четкий и конкретный
              </li>
              <li>
                Только то, что поможет сдать ЕГЭ на 100 баллов
              </li>
              <li>
                Больше повторения (лучше повторять, чем учить то, что на ЕГЭ никогда не попадется)
              </li>
            </ul>
          </div>
        </div>
        <div
          className="landing__advantages__body__box__el"
        >
          <div
            className="landing__advantages__body__box__el__img"
          />
          <div
            className="landing__advantages__body__box__el__text"
          >
            <h2>
              <span>
                Без
              </span>
               
              скучных лекций
            </h2>
            <ul>
              <li>
                На занятиях чередуется теория и практика
              </li>
              <li>
                Лекции ведет заряженный на успех преподаватель, который поделиться с вами не только знаниями, но и жизненным опытом
              </li>
            </ul>
          </div>
        </div>
        <div
          className="landing__advantages__body__box__el"
        >
          <div
            className="landing__advantages__body__box__el__img"
          />
          <div
            className="landing__advantages__body__box__el__text"
          >
            <h2>
              <span>
                Без
              </span>
               
              непонятных объяснений
            </h2>
            <ul>
              <li>
                Мы используем удобные скрипты, что вы не запутались в материале
              </li>
              <li>
                Преподаватель ДОСТУПНЫМ языком объясняет материал НА ПАЛЬЦАХ
              </li>
              <li>
                После занятия вы получаете итоговые конспекты, графические карты, интеллект-схемы для повторения
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="landing__advantages__body__box right"
      >
        <div
          className="landing__advantages__body__box__el"
        >
          <div
            className="landing__advantages__body__box__el__img"
          />
          <div
            className="landing__advantages__body__box__el__text"
          >
            <h2>
              <span>
                Без
              </span>
               
              пустых заданий
            </h2>
            <ul>
              <li>
                Преподаватель ДОСТУПНЫМ языком объясняет материал НА ПАЛЬЦАХ
              </li>
              <li>
                Составляем тесты на основе рекомендованных ФИПИ сборников, а также Банка заданий ФИПИ
              </li>
            </ul>
          </div>
        </div>
        <div
          className="landing__advantages__body__box__el"
        >
          <div
            className="landing__advantages__body__box__el__img"
          />
          <div
            className="landing__advantages__body__box__el__text"
          >
            <h2>
              <span>
                Без
              </span>
               
              возможности схалтурить
            </h2>
            <ul>
              <li>
                У вас есть персональный тьютор, который и пряником угостит, и «пинком» волшебным коли надо одарит
              </li>
              <li>
                У вас есть система жизней, при потере которых вы можете вылететь из группы
              </li>
              <li>
                У вас есть система бонусов, для усердных учеников предусмотрены бесплатные «плюшки» и скидки
              </li>
              <li>
                Каждую неделю проводим коллоквиумы и 1 зачет в месяц – контроль качества знаний обеспечен
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="landing__how-to-study"
  >
    <div
      className="landing__how-to-study__title"
    >
      <h1>
        Как проходят занятия
        <span>
          ?
        </span>
      </h1>
    </div>
    <div
      className="landing__how-to-study-body"
    >
      <div
        className="landing__how-to-study-body__img"
      />
      <div
        className="landing__how-to-study-body__text"
      >
        <p>
          Занятия проходят онлайн. Для каждого занятия есть скрипт, чтобы было проще вести конспект, а информация лучше усваивалась.
        </p>
        <p>
          Длительность: от часа до полутора часов (в зависимости от темы). Начало всегда по московскому времени. Но не переживай, если ты не успел посетить занятие: мы всегда сохраняем видеозапись.
        </p>
        <p>
          Личный тьютор будет помогать тебе, а преподаватель ответит на вопросы
        </p>
      </div>
    </div>
  </div>
  <div
    className="landing__free-webinar"
    id="following"
  >
    <div
      className="landing__free-webinar__title"
    >
      <h1>
        Посети первое занятие бесплатно
      </h1>
    </div>
    <div
      className="landing__free-webinar__body"
    >
      <div
        className="landing__free-webinar__body__left"
      >
        <p>
          Подпишись на нас вКонтакте
          <br />
          и получи приглашение на первое занятие. 
          <br />
           Пора готовиться!
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Подписаться
        </button>
      </div>
      <div
        className="landing__free-webinar__body__right"
      >
        <h1>
          Тебе доступны курсы
        </h1>
        <ul>
          <li>
            Английский язык. ЕГЭ
          </li>
          <li>
            История. ЕГЭ
          </li>
          <li>
            Обществознание. ЕГЭ
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
